src/test/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSparkOptimizedIntegrationTest.java

```diff
@@ -0,0 +1,192 @@
+
+                // Currently disabled because BaseRecalibratorSpark can't handle more than 1 knownSites file.
+                // Re-enable once this is fixed.
+                //{new BQSRTest(GRCh37Ref, HiSeqBam, dbSNPb37, "-knownSites " + moreSites, localResources + "expected.CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.2inputs.recal.txt")},
+
+                {new BQSRTest(GRCh37Ref, HiSeqBam, dbSNPb37, "--indels_context_size 4",  localResources + "expected.CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.indels_context_size_4.recal.txt")},
+                {new BQSRTest(GRCh37Ref, HiSeqBam, dbSNPb37, "--low_quality_tail 5",     localResources + "expected.CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.low_quality_tail_5.recal.txt")},
+                {new BQSRTest(GRCh37Ref, HiSeqBam, dbSNPb37, "--quantizing_levels 6",    localResources + "expected.CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.quantizing_levels_6.recal.txt")},
+                {new BQSRTest(GRCh37Ref, HiSeqBam, dbSNPb37, "--mismatches_context_size 4", localResources + "expected.CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.mismatches_context_size_4.recal.txt")},
+                //// //{new BQSRTest(b36Reference, origQualsBam, dbSNPb36, "-OQ", getResourceDir() + "expected.originalQuals.1kg.chr1.1-1K.1RG.dictFix.OQ.txt")},
+
+                // local reference
+                {new BQSRTest(GRCh37RefLocal, HiSeqBam, dbSNPb37, "", getResourceDir() + "expected.CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.recal.txt")},
+        };
+    }
+
+    @DataProvider(name = "BQSRTestBucket")
+    public Object[][] createBQSRTestDataBucket() {
+        final String GRCh37Ref = ReferenceAPISource.URL_PREFIX + ReferenceAPISource.GRCH37_REF_ID;
+        final String localResources =  getResourceDir();
+        final String HiSeqBamCloud = getCloudInputs() + "CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.bam";
+        final String dbSNPb37 =  getResourceDir() + "dbsnp_132.b37.excluding_sites_after_129.chr17_69k_70k.vcf";
+
+        return new Object[][]{
+                // input in cloud, computation local.
+                {new BQSRTest(GRCh37Ref, HiSeqBamCloud, dbSNPb37, "", localResources + "expected.CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.recal.txt")},
+        };
+    }
+
+
+    // "local", but we're still getting the reference from the cloud.
+    @Test(dataProvider = "BQSRTest", groups = {"cloud"})
+    public void testBQSRLocal(BQSRTest params) throws IOException {
+        ArgumentsBuilder ab = new ArgumentsBuilder().add(params.getCommandLine());
+        IntegrationTestSpec spec = new IntegrationTestSpec(
+                ab.getString(),
+                Arrays.asList(params.expectedFileName));
+        spec.executeTest("testBQSR-" + params.args, this);
+    }
+
+    // TODO: re-enable once ReadsSparkSource natively supports files in GCS buckets
+    @Test(dataProvider = "BQSRTestBucket", groups = {"bucket"}, enabled = false)
```

Comment:
yup. I've done it, though that test takes a long time (probably shouldn't, but does).